probe/appclient/app_client.go

```diff
@@ -231,6 +233,9 @@ func (c *appClient) controlConnection() (bool, error) {
 }
 
 func (c *appClient) ControlConnection() {
+	if c.noControls {
+		return
+	}
```

Comment:
Would it be easier to not plumb this flag through to the appClient, but rather the multiclient, and then not call ControlConnection()?